server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -1319,7 +1319,7 @@ public static StreamInput wrap(byte[] bytes) {
     }
 
     public static StreamInput wrap(byte[] bytes, int offset, int length) {
```

Comment:
Wow we call this method a lot!

I'm curious to see what this change will translate to in terms of raw performance in general use.

Reply:
I think this is a big case of "it depends". The good news is probably that all operations will see a speedup (I ad hoc benchmarked reading strings etc. and there's some speedup there). On the other hand you will not see the single array stream over very large arrays too often (if you did in some spot, allocating those huge arrays should probably be fixed). But I think on balance even if it doesn't speed up too many operations end-to-end, it certainly does improve the load on the transport thread for some medium size messages which is always nice for stability :)